apps/game_backend/lib/game_backend/units.ex

```diff
@@ -75,6 +75,14 @@ defmodule GameBackend.Units do
   """
   def get_units(), do: Repo.all(Unit) |> Repo.preload([:character, :user, :items])
 
+  @doc """
+  Gets all units from ids in a list.
+  """
+  def get_units_by_ids(unit_ids) when is_list(unit_ids),
+    do:
+      Repo.all(from(u in Unit, where: u.id in ^unit_ids))
+      |> Repo.preload([:character, :user, :items])
```

Comment:
The preload should be inside the first query:
```suggestion
      Repo.all(from(u in Unit, where: u.id in ^unit_ids, preload: [:character, :user, :items]))
```